pkg/util/util.go

```diff
@@ -33,20 +33,20 @@ func getConnectSecret(token string) (string, error) {
 }
 
 func ParseSecure(token string) (types.Secure, error) {
-	if len(token) != tokenLength {
-		klog.Fatalf("Token %s length was not %d", token, tokenLength)
-	}
-	secure := types.Secure{}
-	var err error
-	secure.SecretKey, err = getConnectSecret(token)
+	secretKey, err := getConnectSecret(token)
 	if err != nil {
-		klog.Fatalf("Could not parse token to secret")
+		return types.Secure{}, err
```

Comment:
yeah but that's beyond the scope of this patch.

Reply:
Yes, although I agree with marun, it's generally more common to return nil in go libraries for empty/error values. (At least for what I have observed, may need more checking). I guess it's also lighter because the processor doesn't need to zero/init a patch of memory for the struct

Reply:
If we look at https://github.com/rancher/submariner/blob/master/pkg/cableengine/ipsec/ipsec.go#L637

They already use such conventions in other submariner places, may be this is something more we can add into a coding conventions manual for the project.

Reply:
Although, I agree, it's out of the specific scope of this patch, I think it's better to have homogenous return styles.

Reply:
This code base has a mix of return/pass by pointer and value for structs. We need to review all of that...